### PR TITLE
LinkPreview for "/" and "/locale"

### PR DESCRIPTION
**Pull request by vinayaksh42**
Fixes: #312 

Added locale and default locale info to `next.config.js`.

Tried it locally using [ngrok](https://ngrok.com)

### CODE REVIEW

**Comment by ecurrencyhodler** [comment]
Thanks vinayak! So the PR works with:
https://saving-satoshi-lzawodii2-savingsatoshi.vercel.app/en
https://saving-satoshi-lzawodii2-savingsatoshi.vercel.app

But it doesn't with other languages:
https://saving-satoshi-lzawodii2-savingsatoshi.vercel.app/es
https://saving-satoshi-git-fork-vinayaksh42-issue312-savingsatoshi.vercel.app/nl
Screenshot 2023-06-27 at 3 36 04 PM

File: app/layout.tsx [hunk 1]
```diff
@@ -47,6 +47,21 @@ export default function RootLayout({
...
+          content="Code your way through the mysteries of Bitcoin - Saving Satoshi"
+        />
+        <meta
+          property="og:description"
+          content="A fun learning resource for developers to learn about how to build on bitcoin."
+        />
+        <meta
+          property="og:image"
+          content="https://imagedelivery.net/wyrwp3c-j0gDDUWgnE7lig/6818cd1b-ea0d-49a4-ea14-d34c1bbae100/public"
+        />
```
Comment (benalleng) [inline app/layout.tsx:64]: do we want to add any twitter-specific metadata? opengraph suits all of our needs for now though I suppose

File: next.config.js [hunk 2]
```diff
@@ -16,6 +16,10 @@ const nextConfig = {
     })
     return config
   },
+  i18n: {
+    locales: ['en', 'es', 'nl', 'hi'],
+    defaultLocale: 'en',
+  },
```
Comment (benalleng) [inline next.config.js:22]: Strange, this seems to be breaking in the vercel deployment but it doesn't impact me when this change is made locally

File: app/layout.tsx (hunk 1 above)
Comment (vinayaksh42) [reply app/layout.tsx:64]: sure, we can add social media related metadata as well.

File: next.config.js (hunk 2 above)
Comment (vinayaksh42) [reply next.config.js:22]: I just experimented it a bit more and It seems like this change is breaking a lot of other things as well.
such as challenges are not working: https://saving-satoshi-git-fork-vinayaksh42-issue312-savingsatoshi.vercel.app/en/chapters/chapter-2/mining-1?dev=true
Even on localhost the challenges are not working properly. Will look into this issue, until that I am changing this PR to draft.

Comment (benalleng) [reply next.config.js:22]: Does the metadata addition only function if these configs are added as well?

**Comment by ecurrencyhodler** [comment]
Closing this for now.

**Pull request closed without merging**